File: C#/BluffinMuffin.Protocol.DataTypes/SeatInfo.cs
using System.Linq;
using BluffinMuffin.Protocol.DataTypes.Attributes;
using BluffinMuffin.Protocol.DataTypes.Enums;
using Newtonsoft.Json;

namespace BluffinMuffin.Protocol.DataTypes
{
    /// <summary>
    /// 
    /// </summary>
    public class SeatInfo
    {
        /// <summary>
        /// 
        /// </summary>
        [JsonIgnore]
        public bool IsEmpty { get { return Player == null; } }
        /// <summary>
        /// The id of the current seat
        /// </summary>
        [ExampleValue(7)]
        public int NoSeat { get; set; }
        /// <summary>
        /// The information of the player sitting in this seat. If null, there is nobody.
        /// </summary>
        public PlayerInfo Player { get; set; }

        /// <summary>
        /// The attributes of the seat
        /// </summary>
        [ExampleValues(2, SeatAttributeEnum.CurrentPlayer, SeatAttributeEnum.BigBlind)]
        public SeatAttributeEnum[] SeatAttributes { get; set; }
        /// <summary>
        /// 
        /// </summary>
        public SeatInfo()
        {
            SeatAttributes = new SeatAttributeEnum[0];
        }
        /// <summary>
        /// 
        /// </summary>
        /// <returns></returns>
        public SeatInfo Clone()
        {
            if (IsEmpty)
                return new SeatInfo();
            return new SeatInfo()
            {
                Player = Player.Clone(),
                NoSeat = NoSeat,
                SeatAttributes = SeatAttributes,
            };
        }
    }
}

File: Python/bluffinmuffin/protocol/data_types/seat_info.py
from collections import OrderedDict

from bluffinmuffin.protocol.enums import SeatAttributeEnum
from .player_info import PlayerInfo


class SeatInfo:

    def __init__(self, no_seat, player, seat_attributes):
        self.no_seat = no_seat
        self.player = player
        self.seat_attributes = seat_attributes

    def __str__(self):
        return '{0} ({1}) [{2}]'.format(
            self.no_seat,
            self.player,
            ', '.join([SeatAttributeEnum.to_string(x) for x in self.seat_attributes])
        )

    def encode(self):
        d = OrderedDict()
        d['NoSeat'] = self.no_seat
        d['Player'] = self.player.encode()
        d['SeatAttributes'] = [SeatAttributeEnum.to_string(x) for x in self.seat_attributes]
        return d

    @classmethod
    def decode(cls, obj):
        return cls(
            obj["NoSeat"],
            PlayerInfo.decode(obj['Player']),
            [SeatAttributeEnum.parse(x) for x in obj['SeatAttributes']]
        )
